### DISCUSSION TITLE
Image in markdown?

### DISCUSSION THREAD

**Discussion by xdavidchen** [roninro/hugo-theme-puppet#39] [Q&A] [answered]
Does anyone using Puppet theme mind sharing how you add an image to your post for the first time?

I tried using markdown syntax:
` ![](PATH)` and it didn't work.

Using tagging `<img>` didn't work either.

Your advice is appreciated!

**Comment by xdavidchen** [comment]
> ` ![](PATH)` and it didn't work.

After a few tries, it looks like it's not about syntax issue. The markdown syntax would wok if I link to an external online source.

So I guess my problem is with the file structure. Where do I put the local images for the website to render?

Thanks!

**Comment by xdavidchen** [comment, marked as answer]
I ended up solving the problem by creating a "posts" folder under static. And a "titleOfPost" folder under "posts". That way it will generate local images.

[See reference](https://parzival2.github.io/blog/posts/add-images-hugo/)

**Comment by zhenda-hub** [comment]
When I write markdown files(first.md) using editors such as typora and vscode, I usually use relative paths to reference local images. But when putting the markdown file into hugo, in order to display the image. I would set up the project directory structure like this:

content
│
├───posts
│ │ emoji-support.md
│ │ _index.md
│ │
│ └───first
│ │ first.md
│ │
│ └───imgs
│ 1.jpg
│ 2.jpg
│ 3.jpg
│
└───series
└───themes-guide
_index.md

But when I using "hugo -D" to generate the public folder. I got directory structure like this:

public
│ 404.html
├───posts
│ │ index.html
│ │ index.xml
│ │
│ ├───emoji-support
│ │ index.html
│ ├───first
│ │ ├───imgs
│ │ │ 1.jpg
│ │ │ 2.jpg
│ │ │ 3.jpg
│ │ │
│ │ └───first
│ │ index.html

So in order to display image（1.jpg） in hugo web, I need to modify the relative path:
" ./imgs/1.jpg"  -> "../imgs/1.jpg" to show .
